pydantic/tools.py

```diff
@@ -26,9 +27,8 @@ def _get_parsing_type(type_: Any, *, type_name: Optional[NameFactory] = None) ->
 T = TypeVar('T')
 
 
-def parse_obj_as(type_: Type[T], obj: Any, *, type_name: Optional[NameFactory] = None) -> T:
-    model_type = _get_parsing_type(type_, type_name=type_name)  # type: ignore[arg-type]
-    return model_type(__root__=obj).__root__
+def parse_obj_as(type_: Type[T], obj: Any) -> T:
```

Comment:
* Should we retain the `type_name` argument for backwards compatibility and just ignore its value? (I think I'm for this, unless we plan to keep the function around long term, in which case maybe we at least add a deprecation warning for using that argument)
* Should we make some effort to massage the error message into looking _exactly_ the same? (I'm against this)